File: android/examples/android-hello-world/android-hello-world-consumer/app/src/main/java/io/joynr/androidhelloworldconsumer/MainActivity.java
package io.joynr.androidhelloworldconsumer;

import android.os.Bundle;
import android.widget.Button;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;
import androidx.lifecycle.ViewModelProviders;

public class MainActivity extends AppCompatActivity {

    private HelloWorldConsumerViewModel mcvm;

    @Override
    protected void onCreate(final Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        mcvm = ViewModelProviders.of(this).get(HelloWorldConsumerViewModel.class);
        mcvm.init((HelloWorldConsumerApplication) getApplication());

        final Button button = findViewById(R.id.button);

        button.setOnClickListener(view -> mcvm.onClick());
        bindData();
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
    }

    public void bindData() {
        mcvm.onUpdatedString().observe(this, this::updateText);
    }

    private void updateText(final String text) {
        final TextView view = findViewById(R.id.text_box);
        view.setText(text);
    }

}

File: android/examples/android-hello-world-binder/android-hello-world-binder-consumer/app/src/main/java/io/joynr/androidhelloworldbinderconsumer/MainActivity.kt
package io.joynr.androidhelloworldbinderconsumer

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.lifecycle.Observer
import androidx.lifecycle.ViewModelProvider
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    lateinit var consumerViewModel: ConsumerViewModel

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        consumerViewModel = ViewModelProvider(this).get(ConsumerViewModel::class.java)
        consumerViewModel.registerProxy(application)


        talk_button.setOnClickListener {
            consumerViewModel.getString()
        }

        consumerViewModel.providedStr.observe(this, Observer { str ->
            text_box.text = str
        })

    }

}
